From 36d9378b2828d5eb7b57ce065bba4b627fc6ce89 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:28 -0400
Subject: [tools] perf/tests/x86: Fix stack map lookup in dwarf unwind test

Message-id: <1410181752-25631-144-git-send-email-jolsa@redhat.com>
Patchwork-id: 91794
O-Subject: [PATCH RHEL7.1 BZ1134356 143/307] perf tests x86: Fix stack map lookup in dwarf unwind test
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 9a126728165eef9b1e9349fe57b9747ac6c6f7e3
Author: Jiri Olsa <jolsa@kernel.org>
Date: Wed Apr 30 16:39:44 2014 +0200

description
===========
Previous commit 'perf x86: Fix perf to use non-executable stack, again'
moved stack map into MAP__VARIABLE map type again. Fixing the dwarf
unwind test stack map lookup appropriately.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/arch/x86/tests/dwarf-unwind.c b/tools/perf/arch/x86/tests/dwarf-unwind.c
index b8c0102..83bc238 100644
--- a/tools/perf/arch/x86/tests/dwarf-unwind.c
+++ b/tools/perf/arch/x86/tests/dwarf-unwind.c
@@ -23,7 +23,7 @@ static int sample_ustack(struct perf_sample *sample,
 
  sp = (unsigned long) regs[PERF_REG_X86_SP];
 
- map = map_groups__find(&thread->mg, MAP__FUNCTION, (u64) sp);
+ map = map_groups__find(&thread->mg, MAP__VARIABLE, (u64) sp);
  if (!map) {
   pr_debug("failed to get stack map\n");
   free(buf);
-- 
1.7.1